library/include/ck/library/reference_tensor_operation/cpu/reference_conv_fwd.hpp

```diff
@@ -88,13 +88,16 @@ struct ReferenceConvFwd : public device::BaseOperator
                 auto f_ncw = [&](auto n, auto k, auto wo) {
                     float v_acc = 0;
 
-                    for(int c = 0; c < arg.weight_.mDesc.GetLengths()[1]; ++c)
+                    for(int c = 0; c < ck::type_convert<int>(arg.weight_.mDesc.GetLengths()[1]);
+                        ++c)
```

Comment:
I would use size_t here for all indices.

Reply:
I guess there would be hidden size_t -> int then when indexing tensors in the most inner loop

Reply:
HostTensorDescriptor can take size_t as indices. See host_tensor.hpp:103, 264. If we use size_t, the reference operation will not be subject to the 2GB limitation.

Reply:
Addressed by [594a4c2](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/212/commits/594a4c22d129e4eac6930355ea180a0e156735e5)